### ISSUE TITLE
ct-server frequently hangs at startup on watch STH when running in GKE

### ISSUE THREAD

**Issue by AlCutter** [google/certificate-transparency#487]
`docker kill`ing it usually sorts the problem out.
I suspect it might be that the etcd service isn't quite ready yet, but more investigation needed.

**Comment by AlCutter** [comment]
Just had a situation where the only possible explanation is that watches of election proposals was hung, so it looks like this can happen at any time, for any watch, not just at start up.

**Comment by AlCutter** [comment]
May, or may not be related:
I added a sprinkle of logging, and set timeouts on all the etcd HttpConnections to 10secs, and got this on *one* of the 3 jobs:
```
I0211 12:12:02.616636     8 etcd.cc:634] Watch request failed (2: The event in requested index is outdated and cleared), will retry after 5 seconds.
I0211 12:12:03.010799     8 etcd.cc:634] Watch request failed (2: The event in requested index is outdated and cleared), will retry after 5 seconds.
I0211 12:12:03.410933     8 etcd.cc:634] Watch request failed (2: The event in requested index is outdated and cleared), will retry after 5 seconds.
I0211 12:12:03.811383     8 etcd.cc:634] Watch request failed (2: The event in requested index is outdated and cleared), will retry after 5 seconds.
I0211 12:12:07.622915     8 etcd.cc:634] Watch request failed (2: The event in requested index is outdated and cleared), will retry after 5 seconds.
I0211 12:12:08.012758     8 etcd.cc:634] Watch request failed (2: The event in requested index is outdated and cleared), will retry after 5 seconds.
I0211 12:12:08.413593     8 etcd.cc:634] Watch request failed (2: The event in requested index is outdated and cleared), will retry after 5 seconds.
I0211 12:12:08.813704     8 etcd.cc:634] Watch request failed (2: The event in requested index is outdated and cleared), will retry after 5 seconds.
```

Interestingly, the other two still just hang at boot:
```
I0211 12:11:44.598062     9 cert_checker.cc:126] Added 657 new certificate(s) to trusted store
I0211 12:11:44.598951     9 ct-server.cc:240] Found DB with Node UUID: 2b54bc53-1bf7-4d9b-a534-bb2c5f6aee07
I0211 12:11:44.599105     9 ct-server.cc:252] Running in CLUSTERED mode.
I0211 12:11:44.599413     9 etcd.cc:736] EtcdClient: 0x1f08df0
I0211 12:11:44.599448     9 thread_pool.cc:96] ThreadPool starting with 4 threads
I0211 12:11:44.599592     9 etcd.cc:927] EtcdClient::Watch: /root/serving_sth
```

**Comment by AlCutter** [comment]
That `The event in requested index is outdated and cleared` is actually an error returned by etcd it seems.

**Comment by AlCutter** [comment]
It appears that etcd only maintains a watch history of 1000 [etcd] log entries, so watchers watching indices further back than that fail.
See https://github.com/coreos/fleet/issues/408

**Comment by pphaneuf** [comment]
I think this should be fixed now (with commit 10bca189865087ce5d2d5fa0ca256c8d8c33257b), re-open if necessary.
